### ISSUE TITLE
Adding support for reporters

### ISSUE THREAD

**Issue by Kreeg** [prettier/prettier#11833]
Hello!
It will be great to add a support for different reporters, like JUnit. The option to CLI will be someting like `--formatter=prettier-formatter-junit` where `prettier-formatter-junit` will be an external reporter plugin. 

**Environments:**

- Prettier Version: 2.4.1
- Running Prettier via: CLI
- Runtime: Nodejs
- Operating System: CI machine 

**Steps to reproduce:**

```
prettier -l "**/*.{md,css,scss,yaml,yml,ts}"
```

**Expected behavior:**
Added `--formatter=prettier-formatter-junit` option. It also will be great to add the diff first like it mention in #6885 

**Actual behavior:**
No support for reporters

**Comment by duailibe** [comment]
Hi @Kreeg. That sounds like something a 3rd party tool could do, it would just need to capture the output of calling `prettier file` and re-format it.

**Comment by Kreeg** [comment]
@duailibe yeah, maybe you're right, but it cannot be done until #6885 is impemented

**Comment by fisker** [comment]
I'm 👍 to add reporters.
